include/zephyr/bluetooth/mesh/access.h

```diff
@@ -703,18 +712,16 @@ struct bt_mesh_model {
 	/** Model callback structure. */
 	const struct bt_mesh_model_cb * const cb;
 
-#ifdef CONFIG_BT_MESH_MODEL_EXTENSIONS
-	/* Pointer to the next model in a model extension list. */
-	struct bt_mesh_model *next;
-#endif
-
 #ifdef CONFIG_BT_MESH_LARGE_COMP_DATA_SRV
 	/* Pointer to the array of model metadata entries. */
-	struct bt_mesh_models_metadata_entry **metadata;
+	struct bt_mesh_models_metadata_entry ** const metadata;
 #endif
 
+	/** Model ctx variables. */
+	struct bt_mesh_model_ctx * const ctx;
+
 	/** Model-specific user data */
-	void *user_data;
+	void * const user_data;
```

Comment:
Why is the user data restricted with `const` qualifier? Any reason to prohibit customers from more flexible pointer usage?

Reply:
user_data init in model definition

Reply:
Didn't get answer. Could you elaborate?